src/Microsoft.OData.Edm/Csdl/Semantics/CsdlSemanticsVocabularyAnnotation.cs

```diff
@@ -128,7 +128,159 @@ protected IEdmTerm ComputeTerm()
 
         private IEdmExpression ComputeValue()
         {
-            return CsdlSemanticsModel.WrapExpression((this.Annotation).Expression, TargetBindingContext, this.Schema);
+            IEdmTypeReference termType = Term is UnresolvedVocabularyTerm ? null : Term.Type;
+            CsdlExpressionBase adjustedExpression = AdjustStringConstantUsingTermType((this.Annotation).Expression, termType);
+
+            return CsdlSemanticsModel.WrapExpression(adjustedExpression, TargetBindingContext, this.Schema);
+        }
+
+        private static CsdlExpressionBase AdjustStringConstantUsingTermType(CsdlExpressionBase expression, IEdmTypeReference termType)
+        {
+            if (expression == null || termType == null)
+            {
+                return expression;
+            }
+
+            switch (expression.ExpressionKind)
+            {
+                case EdmExpressionKind.Collection:
+                    if (termType.IsCollection())
+                    {
+                        IEdmTypeReference elementType = termType.AsCollection().ElementType();
+                        IList<CsdlExpressionBase> newElements = new List<CsdlExpressionBase>();

```

Comment:
>IList [](start = 24, length = 5)

Why not use List here, instead of IList?

Reply:
I only want to use the "methods" defined in the IList<T>, not all methods defined in List<T>

from List<T> definition as below:
```
class List<T> : ICollection<T>, IEnumerable<T>, IEnumerable, IList<T>, IReadOnlyCollection<T>, IReadOnlyList<T>, ICollection, IList
```
I don't need any methods defined in "ICollection<T>", and so on for others.





---
In reply to: [443006094](https://github.com/OData/odata.net/pull/1775#discussion_r443006094) [](ancestors = 443006094)

Reply:
Not a big deal, but since you have the concrete class thought it might be just as easy to use it.

---
In reply to: [444551603](https://github.com/OData/odata.net/pull/1775#discussion_r444551603) [](ancestors = 444551603,443006094)